plugins/trace_api_plugin/test/test_trace_file.cpp

```diff
@@ -0,0 +1,405 @@
+      next_offset = vs._pos;
+      const auto offset4 = append_data_log( le2, vs );
+      BOOST_REQUIRE_EQUAL(offset4, next_offset);
+
+      vs._pos = offset;
+      const auto be_returned1 = extract_data_log<metadata_log_entry>( vs );
+      BOOST_REQUIRE(be_returned1.contains<block_entry_v0>());
+      const auto real_be_returned1 = be_returned1.get<block_entry_v0>();
+      const auto real_be1 = be1.get<block_entry_v0>();
+      BOOST_REQUIRE(real_be_returned1 == real_be1);
+
+      vs._pos = offset2;
+      const auto le_returned1 = extract_data_log<metadata_log_entry>( vs );
+      BOOST_REQUIRE(le_returned1.contains<lib_entry_v0>());
+      const auto real_le_returned1 = le_returned1.get<lib_entry_v0>();
+      const auto real_le1 = le1.get<lib_entry_v0>();
+      BOOST_REQUIRE(real_le_returned1 == real_le1);
+
+      vs._pos = offset3;
+      const auto be_returned2 = extract_data_log<metadata_log_entry>( vs );
+      BOOST_REQUIRE(be_returned2.contains<block_entry_v0>());
+      const auto real_be_returned2 = be_returned2.get<block_entry_v0>();
+      const auto real_be2 = be2.get<block_entry_v0>();
+      BOOST_REQUIRE(real_be_returned2 == real_be2);
+
+      vs._pos = offset4;
+      const auto le_returned2 = extract_data_log<metadata_log_entry>( vs );
+      BOOST_REQUIRE(le_returned2.contains<lib_entry_v0>());
+      const auto real_le_returned2 = le_returned2.get<lib_entry_v0>();
+      const auto real_le2 = le2.get<lib_entry_v0>();
+      BOOST_REQUIRE(real_le_returned2 == real_le2);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(slice_number, test_fixture)
+   {
+      fc::temp_directory tempdir;
+      boost::filesystem::path tempdir_path = tempdir.path();
+      slice_provider sp(tempdir_path, 100);
+      BOOST_REQUIRE_EQUAL(sp.slice_number(99), 0);
+      BOOST_REQUIRE_EQUAL(sp.slice_number(100), 1);
+      BOOST_REQUIRE_EQUAL(sp.slice_number(1599), 15);
+      slice_provider sp2(tempdir_path, 0x10);
+      BOOST_REQUIRE_EQUAL(sp2.slice_number(0xf), 0);
+      BOOST_REQUIRE_EQUAL(sp2.slice_number(0x100), 0x10);
+      BOOST_REQUIRE_EQUAL(sp2.slice_number(0x233), 0x23);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(slice_file, test_fixture)
+   {
+      fc::temp_directory tempdir;
+      bfs::path tempdir_path = tempdir.path();
+      slice_provider sp(tempdir_path, 100);
+      fc::cfile slice;
+
+      const bool read_file = false;
+      const bool append_file = true;
+      // create trace slices
+      for (uint i = 0; i < 9; ++i) {
+         bool new_file = sp.find_trace_slice(i, append_file, slice);
+         BOOST_REQUIRE(new_file);
+         bfs::path fp = slice.get_file_path();
+         BOOST_REQUIRE_EQUAL(fp.parent_path().generic_string(), tempdir.path().generic_string());
+         const std::string expected_filename = "trace_0000000" + boost::lexical_cast<std::string>(i) + "00-0000000" + boost::lexical_cast<std::string>(i+1) + "00.log";
+         BOOST_REQUIRE_EQUAL(fp.filename().generic_string(), expected_filename);
```

Comment:
I'm on the fence as to whether this is a necessary test.  On the one hand, we expect manual maintenance so, the filesystem structure is an API of sorts.  

On the other hand, this seems like a pedantic test that requires anyone changing that filename to do so in two places.  On top of that it only guards the filename, if someone changes the path structure this test won't catch the breakage of our contract with operators.

Maybe this is something we test for outside of unit tests? where we can actually guarantee the operator-level contract that given a config you will see the following files... ?

Reply:
I don't think useful tests need to stand the test of time and live on forever.  If someone is changing the format, they can decide if this should be removed.  If they don't think about anything, then it is good that we have this here.

Reply:
and having this here, makes that higher level test easier, you just need to make sure that files are read from the correct spot